aptible/metric_drain.go

```diff
@@ -0,0 +1,146 @@
+package aptible
+
+import (
+	"fmt"
+
+	"github.com/aptible/go-deploy/client/operations"
+	"github.com/aptible/go-deploy/models"
+	"github.com/go-openapi/strfmt"
+	"github.com/go-openapi/swag"
+)
+
+type MetricDrain struct {
+	Deleted    bool
+	ID         int64
+	Handle     string
+	DrainType  string
+	DatabaseID int64
+	Address    strfmt.URI
+	Username   string
+	Password   string
+	Database   string
+	APIKey     string
+	SeriesURL  strfmt.URI
+	AccountID  int64
+}
+
+type MetricDrainCreateAttrs struct {
+	DatabaseID   int64
+	DrainType    *string
+	LoggingToken string
+	Address      strfmt.URI
+	Username     string
+	Password     string
+	Database     string
+	APIKey       string
+	SeriesURL    strfmt.URI
+}
+
+func (c *Client) CreateMetricDrain(handle string, accountID int64, attrs *MetricDrainCreateAttrs) (*MetricDrain, error) {
+	if attrs == nil {
+		return nil, fmt.Errorf("no attributes provided")
+	}
+
+	request := &models.AppRequest19{
+		DatabaseID: attrs.DatabaseID,
+		DrainType:  attrs.DrainType,
+		Handle:     &handle,
+	}
+
+	// influxdb_database drains cannot have a DrainConfiguration
+	if attrs.DrainType != nil && *attrs.DrainType != "influxdb_database" {
+		request.DrainConfiguration = &models.AccountsaccountIdmetricDrainsDrainConfiguration{
+			Address:   attrs.Address,
+			Username:  attrs.Username,
+			Password:  attrs.Password,
+			Database:  attrs.Database,
+			APIKey:    attrs.APIKey,
+			SeriesURL: attrs.SeriesURL,
+		}
+	}
+
+	metricDrain := &MetricDrain{}
```

Comment:
probably dont need to do this since you only populate id here. you can return nils everywhere else this is used